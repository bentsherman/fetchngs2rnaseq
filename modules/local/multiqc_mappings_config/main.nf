
process MULTIQC_MAPPINGS_CONFIG {

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    path csv

    output:
    path "*yml"        , emit: yml
    path "versions.yml", emit: versions

    script:
    """
    multiqc_mappings_config.py \\
        $csv \\
        multiqc_config.yml

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
