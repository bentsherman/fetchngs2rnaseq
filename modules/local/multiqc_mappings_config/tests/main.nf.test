nextflow_process {

    name "Test process: MULTIQC_MAPPINGS_CONFIG"
    script "../main.nf"
    process "MULTIQC_MAPPINGS_CONFIG"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'csv/SRX9626017_SRR13191702.mappings.csv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
