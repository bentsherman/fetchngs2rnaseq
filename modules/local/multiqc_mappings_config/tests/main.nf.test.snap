{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "multiqc_config.yml:md5,7f3cb10fff83ba9eb3e8fa6862d1290a",
                        "versions.yml:md5,dd4c66f0551d15510b36bb2e2b2fdd73"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dd4c66f0551d15510b36bb2e2b2fdd73"
                ],
                "versions": [
                    "versions.yml:md5,dd4c66f0551d15510b36bb2e2b2fdd73"
                ],
                "yml": [
                    [
                        "multiqc_config.yml:md5,7f3cb10fff83ba9eb3e8fa6862d1290a",
                        "versions.yml:md5,dd4c66f0551d15510b36bb2e2b2fdd73"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T11:52:12.65888"
    }
}